keps/sig-storage/1847-autoremove-statefulset-pvcs/README.md

```diff
@@ -0,0 +1,514 @@
+you're proposing, but should not include things like API designs or
+implementation.  The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system.  The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+User environment is such at the content of the PVCs which are created automatically during StatefulSet 
+creation need not be retained after the StatefulSet is deleted. User also requires that the scale 
+up/down occurs in a fast manner, and leverages any previously existing auto created PVCs within the 
+life time of the StatefulSet. An option needs to be provided for the user to auto-delete the PVCs 
+once the StatefulSet is deleted. 
+
+User would set the VolumeReclaimPolicy as 'RemoveOnStatefulSetDelete' which would ensure that 
+the PVCs created automatically during the StatefulSet activation is removed once the StatefulSet 
+is deleted.
+
+#### Story 2
+User is cost conscious but at the same time can sustain slower scale up(after a scale down) speeds. Needs 
+a provision where the PVC created for a pod(which is part of the StatefulSet) is removed when the Pod 
+is deleted as part of a scale down. Since the subsequent scale up needs to create fresh PVCs, it will 
+be slower than scale ups relying on existing PVCs(from earlier scale ups). 
+
+User would set the VolumeReclaimPolicy as 'RemoveOnScaledown' ensuring PVCs are deleted when corresponding
+Pods are deleted. New Pods created during sclae up followed by a scaledown will wait for freshly created PVCs.
+
+### Notes/Constraints/Caveats (optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above.
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal and how do we mitigate.  Think broadly.
+For example, consider both security and how this will impact the larger
+kubernetes ecosystem.
+
+How will security be reviewed and by whom?
+
+How will UX be reviewed and by whom?
+
+Consider including folks that also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable.  This may include API specs (though not always
+required) or even code snippets.  If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+### Volume reclaim policy for the StatefulSet created PVCs
+A new field named `VolumeReclaimPolicy` of the type `StatefulSetVolumeReclaimPolicy` 
+will be added to the StatefulSet. This field represents the user indication on whether the 
```

Comment:
Added a note to ensure that in alpha we won't allow updating this field after creation to avoid any confusion. Based user feedback we can change this in beta.